### DISCUSSION TITLE
ltsp image error

### DISCUSSION THREAD

**Discussion by ghost** [ltsp/ltsp#319] [Q&A] [answered]
(warning English is not my mother tong) Hi I'm following the documentation on https://ltsp.org/docs/installation/, I'm using an qemu VM to create my client image, The first thing i tried was to convert my VM (kubuntu 20.04) from qcow2 format to raw (sudo qemu-img convert -f qcow2 -O raw image.qcow2 image.img).

Then when i execute ltsp image image.img i got the following error : 
sudo ltsp image ubuntu.img 
Running: losetup -rP /dev/loop0 /srv/ltsp/ubuntu.img
Running: mount -t tmpfs -o mode=0755 tmpfs /tmp/tmp.e3mJxzxCkr/tmpfs
Running: mount -t ext4 -o ro,noload /dev/loop0p1 /tmp/tmp.e3mJxzxCkr/tmpfs/0/looproot
Running: mount -t overlay -o upperdir=/tmp/tmp.e3mJxzxCkr/tmpfs/0/up,lowerdir=/tmp/tmp.e3mJxzxCkr/tmpfs/0/looproot,workdir=/tmp/tmp.e3mJxzxCkr/tmpfs/0/work /tmp/tmp.e3mJxzxCkr/tmpfs /tmp/tmp.e3mJxzxCkr/root/
Cleaning up ubuntu before mksquashfs...
Traceback (most recent call last):
  File "/usr/share/ltsp/client/login/pwmerge", line 440, in <module>
    main(sys.argv)
  File "/usr/share/ltsp/client/login/pwmerge", line 434, in main
    pwm = PwMerge(args[0], args[1], args[2], **dopts)
  File "/usr/share/ltsp/client/login/pwmerge", line 117, in __init__
    self.dpasswd, self.dgroup = self.read_dir(ddir, dur or dgr)
  File "/usr/share/ltsp/client/login/pwmerge", line 151, in read_dir
    with open("{}/passwd".format(xdir), "r") as file:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp.e3mJxzxCkr/root/etc/passwd'
LTSP command failed: /usr/share/ltsp/client/login/pwmerge --ltsp --quiet /tmp/tmp.e3mJxzxCkr/root/tmp/pwempty /tmp/tmp.e3mJxzxCkr/root/etc /tmp/tmp.e3mJxzxCkr/root/tmp/pwmerged
Aborting ltsp

I tried to create a new VM from scratch using the raw format, but when I tried again I got another error : 
sudo ltsp image ubuntu.img 
Running: losetup -rP /dev/loop0 /srv/ltsp/ubuntu.img
Running: mount -t tmpfs -o mode=0755 tmpfs /tmp/tmp.LVFKN4uVn6/tmpfs
Running: mount -t vfat -o ro /dev/loop0p1 /tmp/tmp.LVFKN4uVn6/tmpfs/0/looproot
Running: mount -t overlay -o upperdir=/tmp/tmp.LVFKN4uVn6/tmpfs/0/up,lowerdir=/tmp/tmp.LVFKN4uVn6/tmpfs/0/looproot,workdir=/tmp/tmp.LVFKN4uVn6/tmpfs/0/work /tmp/tmp.LVFKN4uVn6/tmpfs /tmp/tmp.LVFKN4uVn6/root/
mount: /tmp/tmp.LVFKN4uVn6/root: wrong fs type, bad option, bad superblock on /tmp/tmp.LVFKN4uVn6/tmpfs, missing codepage or helper program, or other error.
LTSP command failed: mount -t overlay -o upperdir=/tmp/tmp.LVFKN4uVn6/tmpfs/0/up,lowerdir=/tmp/tmp.LVFKN4uVn6/tmpfs/0/looproot,workdir=/tmp/tmp.LVFKN4uVn6/tmpfs/0/work /tmp/tmp.LVFKN4uVn6/tmpfs /tmp/tmp.LVFKN4uVn6/root/
Aborting ltsp

Any ideas on what I may be missing  ? 

Thanks.

**Comment by alkisg** [comment, marked as answer]
Hi, I think this discussion is the same as this one: https://github.com/ltsp/ltsp/discussions/295

**Reply by ghost** [reply]
Yes, I create a new image using manual partition and it worked, thanks.
